### ISSUE TITLE
Error compiling  hyprland 0.40.0 on Venom linux

### ISSUE THREAD

**Issue by cngr** [hyprwm/Hyprland#5898]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
<Paste the output of the command here>
```

</details>

### Bug or Regression?

Bug

### Description

I never had problems to update hyprland in venom linux but now a have this problem

### How to reproduce

the new update need some patch to compile?

### Crash reports, logs, images, videos

![imagen](https://github.com/hyprwm/Hyprland/assets/10997809/733d7e5b-9c0e-431f-b804-564497a4663e)

**Comment by Agent00Ming** [comment]
Can you build Hyprland manually? Seems like your distro's packaging solution broke something with regards to the patches and their application.

**Comment by cngr** [comment]
No i tried manually and have same problem

**Comment by Agent00Ming** [comment]
what does running `./patches/apply.sh` in `subprojects/wlroots-hyprland` output?

**Comment by cngr** [comment]
As you can see in the screenshot it gives an error when it reaches the patch, manually it has the same error, in previous versions of hyprland it compiled perfectly and everything works ok, it was only in this version

**Comment by Agent00Ming** [comment]
If you ran the script in manually or check the meson logs in `subproject/wlroots/build/meson-logs/meson.log` you would know which patch failed.

**Comment by cngr** [comment]
Thank you very much for commenting, it has made the change that you said and now it compiles well for me, they should change it to be able to update the port in Venom, but I already have it updated, thank you

**Comment by jetm** [comment]
Patch https://github.com/hyprwm/wlroots-hyprland/pull/11 from @Agent00Ming works, but after it is applied, another build error happens:
```
FAILED: CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o
/usr/bin/c++ -DHAS_EXECINFO -DHyprland_EXPORTS -DUSES_SYSTEMD -DWLR_USE_UNSTABLE -I/Hyprland-0.40.0/.-I/Hyprland-0.40.0/src -I/Hyprland-0.40.0/subprojects/wlroots-hyprland/include -I/Hyprland-0.40.0/subprojects/wlroots-hyprland/build/include -I/Hyprland-0.40.0/subprojects/udis86 -I/Hyprland-0.40.0/protocols -I/Hyprland-0.40.0/subprojects/udis86/libudis86 -isystem /usr/include/uuid -isystem /usr/include/pango-1.0 -isystem /usr/include/cairo -isystem /usr/include/glib-2.0 -isystem /usr/lib/glib-2.0/include -isystem /usr/include/harfbuzz -isystem /usr/include/freetype2 -isystem /usr/include/pixman-1 -isystem /usr/include/libdrm -isystem /usr/include/libpng16 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/sysprof-6 -isystem /usr/include/fribidi -march=native -mtune=native -O3 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -flto=auto -O3 -DNDEBUG -std=gnu++23 -O3 -Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing -Wno-pointer-arith -pthread -Winvalid-pch -include /Hyprland-0.40.0/build/CMakeFiles/Hyprland.dir/cmake_pch.hxx -MD -MT CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o-MF CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o.d -o CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o -c /Hyprland-0.40.0/src/protocols/OutputManagement.cpp
/Hyprland-0.40.0/src/protocols/OutputManagement.cpp: In member function ‘void COutputHead::sendAllData()’:
/Hyprland-0.40.0/src/protocols/OutputManagement.cpp:169:64: error: cannot convert ‘wl_resource*’ to ‘CZwlrOutputModeV1*’
  169 |                 resource->sendCurrentMode(m->resource->resource());
      |                                           ~~~~~~~~~~~~~~~~~~~~~^~
      |                                                                |
      |                                                                wl_resource*
```

**Comment by vaxerski** [comment]
you updated hyprwayland-scanner but didn't update hyprland

**Comment by jetm** [comment]
@vaxerski Yes, it is built now after the pushed commits today. Thank you for the quick response.

**Comment by lfos** [comment]
@vaxerski While pulling in latest changes addresses the error, isn't the latest release tarball is still broken? It does not include ec092bd. Should there be a bugfix release 0.4.1?

**Comment by vaxerski** [comment]
it's not broken. 0.40.0 requires hw-s = 0.3.4

**Comment by lfos** [comment]
@vaxerski I was not referring to the dependency; I was referring to the build error pointed out by @jetm later in this thread:

```
FAILED: CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o
/usr/bin/c++ -DHAS_EXECINFO -DHyprland_EXPORTS -DUSES_SYSTEMD -DWLR_USE_UNSTABLE -I/Hyprland-0.40.0/.-I/Hyprland-0.40.0/src -I/Hyprland-0.40.0/subprojects/wlroots-hyprland/include -I/Hyprland-0.40.0/subprojects/wlroots-hyprland/build/include -I/Hyprland-0.40.0/subprojects/udis86 -I/Hyprland-0.40.0/protocols -I/Hyprland-0.40.0/subprojects/udis86/libudis86 -isystem /usr/include/uuid -isystem /usr/include/pango-1.0 -isystem /usr/include/cairo -isystem /usr/include/glib-2.0 -isystem /usr/lib/glib-2.0/include -isystem /usr/include/harfbuzz -isystem /usr/include/freetype2 -isystem /usr/include/pixman-1 -isystem /usr/include/libdrm -isystem /usr/include/libpng16 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/sysprof-6 -isystem /usr/include/fribidi -march=native -mtune=native -O3 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -flto=auto -O3 -DNDEBUG -std=gnu++23 -O3 -Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing -Wno-pointer-arith -pthread -Winvalid-pch -include /Hyprland-0.40.0/build/CMakeFiles/Hyprland.dir/cmake_pch.hxx -MD -MT CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o-MF CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o.d -o CMakeFiles/Hyprland.dir/src/protocols/OutputManagement.cpp.o -c /Hyprland-0.40.0/src/protocols/OutputManagement.cpp
/Hyprland-0.40.0/src/protocols/OutputManagement.cpp: In member function ‘void COutputHead::sendAllData()’:
/Hyprland-0.40.0/src/protocols/OutputManagement.cpp:169:64: error: cannot convert ‘wl_resource*’ to ‘CZwlrOutputModeV1*’
  169 |                 resource->sendCurrentMode(m->resource->resource());
      |                                           ~~~~~~~~~~~~~~~~~~~~~^~
      |                                                                |
      |                                                                wl_resource*
```

This build error seems to be fixed by ec092bd. When I built the hyprland package for Arch Linux, in addition to adding the new dependency, I had to manually pull in and apply ec092bd to the release tarball in order to get a successful build. Thus my proposal to create a bugfix release. Am I wrong or missing something here?

**Comment by lfos** [comment]
> This build error seems to be fixed by [ec092bd](https://github.com/hyprwm/Hyprland/commit/ec092bd601d9d351ff6ca34bd97f12055b2a4dd9). When I built the hyprland package for Arch Linux, in addition to adding the new dependency, I had to manually pull in and apply [ec092bd](https://github.com/hyprwm/Hyprland/commit/ec092bd601d9d351ff6ca34bd97f12055b2a4dd9) to the release tarball in order to get a successful build. Thus my proposal to create a bugfix release. Am I wrong or missing something here?

Nevermind, I missed the `= 0.3.4` part from your comment; thanks for pointing that out. That's quite subtle, hopefully there will be better mechanisms to keep things compatible between different versions (or exact dependencies more clear) going forward.
